src/Tests/BusinessLayer/TelemetryReportingTest.cpp

```diff
@@ -0,0 +1,959 @@
+        }
+
+        virtual void DescribeTo(::std::ostream* os) const
+        {
+            *os << "Expected package ID = " << expectedId_ << std::endl;
+        }
+
+    private:
+        const unsigned char expectedId_;
+        const TelemetryReportingTest& containingTest_;
+    };
+
+    inline Matcher<std::tuple<const unsigned char*, int>> packageIdIs(unsigned char expectedId)
+    {
+        return MakeMatcher(new PackageIdMatcher(expectedId, *this));
+    }
+
+};
+
+/*
+ * Actual test cases are starting here
+ */
+
+/*
+ * This test tests for the correct structure of the sendKeyMotor package as defined in
+ * https://docs.google.com/spreadsheets/d/1soVLjeD9Sl7z7Z6cYMyn1fmn-cG7tx_pfFDsvgkCqMU/edit#gid=782574835
+ *
+ * The stuffing, framing and conversion is assumed to work correctly here. These methods are tested
+ * separately.
+ */
+TEST_F(TelemetryReportingTest, sendKeyMotorTest) // TODO create function which build the actual package to create more test cases in a easy way ...
+{
+    // prepare payload
+    const unsigned int expectedPackageLength = EXPECTED_PACKAGE_LENGTH_SEND_KEY_MOTOR;
+    const unsigned int payloadLength = expectedPackageLength - COBS_ADDITIONAL_FRAME_DATA_SIZE;
+    unsigned char data[payloadLength];
+    data[0] = CcsDefines::KEY_MOTOR_PKG_ID;
+    bool motor0Alive[] = {keyMotorData_->motor0Alive};
+    telemetryReporting_->writeBoolsIntoArray(data, 1, motor0Alive, 1);
+    telemetryReporting_->writeFloatIntoArray(data, 2, keyMotorData_->motor0SetCurrent);
+    telemetryReporting_->writeFloatIntoArray(data, 6, keyMotorData_->motor0SetVelocity);
+    telemetryReporting_->writeFloatIntoArray(data, 10, keyMotorData_->motor0BusCurrent);
+    telemetryReporting_->writeFloatIntoArray(data, 14, keyMotorData_->motor0BusVoltage);
+    telemetryReporting_->writeFloatIntoArray(data, 18, keyMotorData_->motor0VehicleVelocity);
+    bool motor1Alive[] = {keyMotorData_->motor1Alive};
+    telemetryReporting_->writeBoolsIntoArray(data, 22, motor1Alive, 1);
+    telemetryReporting_->writeFloatIntoArray(data, 23, keyMotorData_->motor1SetCurrent);
+    telemetryReporting_->writeFloatIntoArray(data, 27, keyMotorData_->motor1SetVelocity);
+    telemetryReporting_->writeFloatIntoArray(data, 31, keyMotorData_->motor1BusCurrent);
+    telemetryReporting_->writeFloatIntoArray(data, 35, keyMotorData_->motor1BusVoltage);
+    telemetryReporting_->writeFloatIntoArray(data, 39, keyMotorData_->motor1VehicleVelocity);
+    appendChecksum(data, payloadLength);
+    // do some additional data checks
+    ASSERT_THAT(data[0], Eq(0x01)); // packet id
+    ASSERT_THAT(data[1], Eq(0x01));
+    ASSERT_THAT(data[22], Eq(0x00));
+    unsigned char expectedPacket[expectedPackageLength];
+    telemetryReporting_->frameData(data, payloadLength, expectedPacket);
+    // check call
+    EXPECT_CALL(*communicationService_, sendData(_, expectedPackageLength)).With(Args<0, 1>(ElementsAreArray(expectedPacket, expectedPackageLength))).Times(1);
```

Comment:
although it is a pretty long line I still think it is quite readable because of the powerful google test matcher mechanism.
However I extracted a part into a local variable splitting the one liner into two shorter parts